Assets/GameAssets/Player/PlayerScripts/PlayerInteraction.cs

```diff
@@ -0,0 +1,108 @@
+/**********************************************************************************************************************
+// File Name :         PlayerInteraction.cs
+// Author :            Alex Kalscheur
+// Creation Date :     11/10/24
+// 
+// Brief Description : Controls the interaction system on the player side
+**********************************************************************************************************************/
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.InputSystem;
+
+[RequireComponent(typeof(PlayerInput))]
+
+/// <summary>
+/// Controls the interaction system on the player side
+/// </summary>
+public class PlayerInteraction : MonoBehaviour
+{
+    [Tooltip("How far can the player reach interactable objects")]
+    [SerializeField] private float _maxReach;
+    [Tooltip("Ray that is cast from camera to find interactable objects")]
+    Ray ray;
+
+    //Input
+    private PlayerInput _playerInput;
+    private InputAction _interactInput;
+
+
+    /// <summary>
+    /// Every frame, Ray is cast.  If the ray hits something and the button is
+    /// </summary>
+    private void Update()
+    {
+        ray = Camera.main.ViewportPointToRay(new Vector3(0.5f, 0.5f, 0));
+        if (Physics.Raycast(ray, out RaycastHit hit, _maxReach))
+        {
+            GiveInteractableFeedback(hit.collider.gameObject);
+            if (_interactInput.WasPerformedThisFrame()) 
+
+            {
+                Interact(hit.collider.gameObject);
+            }
+        }
+    }
+
+    /// <summary>
+    /// checks if gameObject can be interacted with
+    ///     If so, calls GiveInteractableFeedback, which should give player feedback that the object is interactable
+    /// </summary>
+    /// <param name="gameObject">GameObject that can potentially be interacted with</param>
+    private void GiveInteractableFeedback(GameObject gameObject)
+    {
+        if (gameObject.TryGetComponent(out IPlayerInteractable interactable))
+        {
+            //REPLACE THIS WHEN WE MAKE UI FOR INTERACTION
+            Debug.Log("Press E to Interact");
```

Comment:
Perhaps the Interactable could have a function for when it is hovered over so that they can display different things?

Reply:
I think this is fine for now and if we want to add this in later we can